db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java

```diff
@@ -994,6 +994,9 @@ public SchemaDescriptor getSchemaDescriptor(String schemaName,
         if(tc==null){
             tc=getTransactionCompile();
         }
+        if (dbId == null) {
+            dbId = getLCC().getDatabaseId();
+        }
 
         if(getSystemSchemaDescriptor().getSchemaName().equals(schemaName)){
```

Comment:
We should check not only SysSchemaDescriptor and SysIBMSchemaDescriptor here, but also the below list of system schemas:
```
    protected SchemaDescriptor sysIBMADMSchemaDesc;
    protected SchemaDescriptor systemUtilSchemaDesc;
    protected SchemaDescriptor sysFunSchemaDesc;
    protected SchemaDescriptor sysViewSchemaDesc;
```
Otherwise, we cannot query objects in these schemas from other databases.

Reply:
Done